libsolidity/ast/Types.cpp

```diff
@@ -1698,6 +1698,8 @@ bool ArrayType::operator==(Type const& _other) const
 
 bool ArrayType::validForCalldata() const
 {
+	if (auto arrayBaseType = dynamic_cast<ArrayType const*>(baseType().get()))
```

Comment:
Wouldn't it be better to call `arrayBaseType->validForCalldata()` here?
A counterexample could be `uint[][2**30][2**30]`

Reply:
Not calling ``arrayBaseType->validForCalldata()`` would still result in an assertion for ``function f(uint[2**30][2**30][] memory) public pure {}``. The same code does not assert when ``ABIEncoderV2``is enabled.